src/main/scala/scorex/core/network/PeerSpec.scala

```diff
@@ -45,7 +45,7 @@ class PeerSpecSerializer(featureSerializers: PeerFeature.Serializers) extends Sc
 
 
     w.putOption(obj.declaredAddress) { (writer, isa) =>
-      val addr = isa.getAddress.getAddress
+      val addr = if(isa.getAddress == null) Array.fill[Byte](4){0} else isa.getAddress.getAddress
```

Comment:
Why to write 0s ? I guess it is safer to write None in putOption (thus trying to decode address before putOption)

Reply:
@jellymlg looks good now. Any known test vectors for getting null in address to make a test for this case ?

Reply:
> @jellymlg looks good now. Any known test vectors for getting null in address to make a test for this case ?

From what I can tell, this can happen when the connecting peer is running in the cloud and has an unresolveable declaredAddress (just an idea)